src/media/manage/MediaSyncElement.ts

```diff
@@ -0,0 +1,214 @@
+import debounce from 'just-debounce-it';
+import { html, LitElement } from 'lit';
+import { property } from 'lit/decorators.js';
+
+import {
+  DisposalBin,
+  hostedEventListener,
+  listen,
+  vdsEvent
+} from '../../base/events';
+import type { VolumeChange, VolumeChangeEvent } from '../events';
+import type { MediaProviderElement } from '../provider';
+
+const mediaProviders = new Set<MediaProviderElement>();
+
+let syncingMediaPlayback = false;
+let syncingMediaVolume = false;
+
+/**
+ * This element is responsible for synchronizing elements of the type `MediaProviderElement`.
+ *
+ * Synchronization includes:
+ *
+ * - Shared media playback (eg: user plays a video while another is already playing, so we pause
+ * the newly inactive player).
+ *
+ * - Shared media volume (eg: user sets desired volume to 50% on one player, and they expect it to
+ * be consistent across all players).
+ *
+ * - Saving media volume to local storage (eg: user sets derived to volume 50%, they leave
+ * the site, and when they come back they expect it to be 50% without any interaction).
```

Comment:
I wonder if this should be local storage vs something more like a session cookie, but this seems fine

Reply:
Ye that could be useful in some cases. This optional anyway, so we can leave other approaches as an application layer concern.